packages/dialog/test/dialog-wrapper.test.ts

```diff
@@ -290,27 +293,82 @@ describe('Dialog Wrapper', () => {
-        const consoleWarnStub = stub(console, 'warn');
-        const el = await fixture<DialogWrapper>(html`
-            <sp-dialog-wrapper></sp-dialog-wrapper>
-        `);
-
-        await elementUpdated(el);
+    describe('dev mode', () => {
+        let consoleWarnStub!: SinonStub;
+        before(() => {
+            consoleWarnStub = stub(console, 'warn');
+        });
+        afterEach(() => {
+            consoleWarnStub.resetHistory();
+        });
+        after(() => {
+            consoleWarnStub.restore();
+        });
+        it('warns in Dev Mode when accessible attributes are not leveraged', async () => {
+            const el = await fixture<DialogWrapper>(html`
+                <sp-dialog-wrapper></sp-dialog-wrapper>
+            `);
+
+            await elementUpdated(el);
+
+            expect(consoleWarnStub.called).to.be.true;
+            const spyCall = consoleWarnStub.getCall(0);
+            expect(
+                spyCall.args.at(0).includes('accessible'),
+                'confirm accessibility-centric message'
+            ).to.be.true;
+            expect(spyCall.args.at(-1), 'confirm `data` shape').to.deep.equal({
+                data: {
+                    localName: 'sp-dialog-wrapper',
+                    type: 'accessibility',
+                    level: 'default',
+                },
+            });
+        });
+    });
 
-        expect(consoleWarnStub.called).to.be.true;
-        const spyCall = consoleWarnStub.getCall(0);
-        expect(
-            spyCall.args.at(0).includes('accessible'),
-            'confirm accessibility-centric message'
-        ).to.be.true;
-        expect(spyCall.args.at(-1), 'confirm `data` shape').to.deep.equal({
-            data: {
-                localName: 'sp-dialog-wrapper',
-                type: 'accessibility',
-                level: 'default',
-            },
+    it('manages content element tabindex on resize observer time', async () => {
+        const imgReadyPromise = new Promise((res) => {
+            const img = document.createElement('img');
+            img.onload = res;
+            img.src = lazyHero.args.src;
+        });
+        const test = await styledFixture(lazyHero(lazyHero.args));
+        const dialog = document.querySelector(
+            'sp-dialog-wrapper'
+        ) as DialogWrapper;
+        const button = document.querySelector('sp-button') as Button;
+        const rect = button.getBoundingClientRect();
+        const contentElement = (
+            (dialog as unknown as { dialog: Dialog }).dialog as unknown as {
+                contentElement: HTMLElement;
+            }
+        ).contentElement;
+        expect(contentElement.hasAttribute('tabindex')).to.be.false;
+        await elementUpdated(dialog);
+        // Lazily updated render content timing.
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
```

Comment:
NIT: Why do you need it 4 times? Maybe better to provide a timeout with large enough time? Or just leave a more detailed comment.